### ISSUE TITLE
UserFileCache.removeFile throws on success

### ISSUE THREAD

**Issue by cbbrainerd** [dmwm/WMCore#9594]
**Impact of the bug**
CRAB

**Describe the bug**
UserFileCache.removeFile throws TypeError on success.

**How to reproduce it**
```crab purge -d <CRAB Directory>```

Alternatively, without crab:

```
from WMCore.Services.UserFileCache.UserFileCache import UserFileCache
ufc=UserFileCache({'endpoint': u'https://cmsweb.cern.ch/crabcache', 'pycurl': True})
hashkey=ufc.upload('some_file.tgz')['hashkey']
ufc.removeFile(hashkey)
```

**Expected behavior**
```UserFileCache.removeFile``` returns successfully.

**Additional context and error message**
Traceback:
```
ufc.removeFile throws
Traceback (most recent call last):
  File "purge_test.py", line 10, in <module>
    res=crab('purge',dir=sys.argv[1])
  File "/cvmfs/cms.cern.ch/crab3/slc6_amd64_gcc493/cms/crabclient/3.3.2001/lib/python2.7/site-packages/CRABAPI/RawCommand.py", line 26, in crabCommand
    return execRaw(command, arguments)
  File "/cvmfs/cms.cern.ch/crab3/slc6_amd64_gcc493/cms/crabclient/3.3.2001/lib/python2.7/site-packages/CRABAPI/RawCommand.py", line 46, in execRaw
    res = cmdobj()
  File "/cvmfs/cms.cern.ch/crab3/slc6_amd64_gcc493/cms/crabclient/3.3.2001/lib/python2.7/site-packages/CRABClient/Commands/purge.py", line 65, in __call__
    ufcresult = ufc.removeFile(hashkey)
  File "/cvmfs/cms.cern.ch/crab3/slc6_amd64_gcc493/cms/crabclient/3.3.2001/lib/python2.7/site-packages/WMCore/Services/UserFileCache/UserFileCache.py", line 123, in removeFile
    return result[0]['result'][0]
TypeError: string indices must be integers, not str
```

**Comment by amaltaro** [comment]
Christopher, thanks for filing this bug and properly describing it.

**Comment by cbbrainerd** [comment]
Fixed by #9595
